### DISCUSSION TITLE
How to bundle an .so file?

### DISCUSSION THREAD

**Discussion by fcoury** [tauri-apps/tauri#6409] [General]
Is there a way to distribute an .so file alongside my app?

I tried:

```
      "resources": [
        "librazermacos.so"
      ],
```

And also `bundle` > `externalBin` but I am always getting this error when running my app:

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: librazermacos.so
Referenced from: <D04CC4EE-C0CA-33E6-87FB-91DD92426416> /Applications/RazerMon.app/Contents/MacOS/RazerMon
Reason: tried: 'librazermacos.so' (relative path not allowed in hardened program), '/System/Volumes/Preboot/Cryptexes/OSlibrazermacos.so' (no such file), 'librazermacos.so' (relative path not allowed in hardened program), '/usr/lib/librazermacos.so' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)

Here's how my app structure looks:

<img width="235" alt="image" src="https://user-images.githubusercontent.com/1371/223487154-da9df75a-b39b-4105-a6c7-a368dd154d3b.png">

You can see that the .so file is right besides the "executable".

Any idea how to either install the .so file as part of the installation process or make it so that it recognizes the file?

**Comment by jiiihpeeh** [comment]
Have you tried to make a static library. It should suck the static lib inside the main  binary.
Here is a project where I did exactly that:
https://github.com/jiiihpeeh/erroshavelimitstoo-Tauri/tree/main/src-tauri

**Comment by x93008** [comment]
Is there a solution to this problem？
